src/refract/Element.cc

```diff
@@ -129,4 +131,276 @@ namespace refract
         }
     }
 
+    namespace {
+        struct TypeChecker {
```

Comment:
Will receive container of elements via ref. Later `bool` typecasting operator will check, if all elements in container are of same type. 

It is alternative way to check it in c-tor and save bool value as result, and later just return result. Or it can be implemnted just as simple function.

I mean that was more complicated while a start it write and in a few iterations I refactor it into this.

In this case it is not important which way is used because we call `bool` casting operator just once on line 164 `if (!TypeChecker(val)) { // ...`